### PR TITLE
Added new property to hide oncoprint menu settings (rebased)

### PR DESCRIPTION
**Pull request by dianab0**
Added new portal.properties property to hide oncoprint
menu settings "cBioPortal>=" and "COSMIC>=";

Backend PR: cBioPortal/cbioportal#9220

### CODE REVIEW

File: src/config/IAppConfig.ts [hunk 1]
```diff
@@ -66,6 +66,7 @@ export interface IServerConfig {
     api_cache_limit: number;
     show_hotspot: boolean | undefined;
     show_oncokb: boolean;
+    show_oncokb_cbiocosmic: boolean;
```
Comment (pvannierop) [inline src/config/IAppConfig.ts:69]: Update property names after update of backend branch.

File: src/config/serverConfigDefaults.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ const ServerConfigDefaults: Partial<IServerConfig> = {
     isoformOverrideSource: 'uniprot',
     show_hotspot: true,
     show_oncokb: true,
+    show_oncokb_cbiocosmic: true,
```
Comment (pvannierop) [inline src/config/serverConfigDefaults.ts:45]: Update property names after update of backend branch.

File: src/shared/components/driverAnnotations/DriverAnnotationControls.tsx [hunk 3]
```diff
@@ -244,69 +245,74 @@ export default class DriverAnnotationControls extends React.Component<
...
                                 )}
-                            {this.props.handlers
-                                .onChangeAnnotateCBioPortalInputValue && (
-                                <div className="checkbox">
-                                    <label>
-                                        <input
-                                            type="checkbox"
-                                            value={EVENT_KEY.annotateCBioPortal}
-                                            checked={
+                            {getServerConfig().show_oncokb_cbiocosmic &&
```
Comment (pvannierop) [inline src/shared/components/driverAnnotations/DriverAnnotationControls.tsx:248]: Update property names after update of backend branch.

File: src/shared/components/driverAnnotations/DriverAnnotationControls.tsx [hunk 4]
```diff
@@ -244,69 +245,74 @@ export default class DriverAnnotationControls extends React.Component<
...
-                                .onChangeAnnotateCOSMICInputValue && (
-                                <div className="checkbox">
-                                    <label>
-                                        <input
-                                            type="checkbox"
-                                            value={EVENT_KEY.annotateCOSMIC}
-                                            checked={
+                                    </div>
+                                )}
+                            {getServerConfig().show_oncokb_cbiocosmic &&
```
Comment (pvannierop) [inline src/shared/components/driverAnnotations/DriverAnnotationControls.tsx:283]: Update property names after update of backend branch.

**Review by pvannierop — CHANGES_REQUESTED** [review]
Update property names after update of corresponding backend branch.

File: src/config/IAppConfig.ts (hunk 1 above)
Comment (dianab0) [reply src/config/IAppConfig.ts]: fixed

File: src/config/serverConfigDefaults.ts (hunk 2 above)
Comment (dianab0) [reply src/config/serverConfigDefaults.ts]: fixed

File: src/shared/components/driverAnnotations/DriverAnnotationControls.tsx (hunk 3 above)
Comment (dianab0) [reply src/shared/components/driverAnnotations/DriverAnnotationControls.tsx]: fixed

File: src/shared/components/driverAnnotations/DriverAnnotationControls.tsx (hunk 4 above)
Comment (dianab0) [reply src/shared/components/driverAnnotations/DriverAnnotationControls.tsx]: fixed

File: src/config/serverConfigDefaults.ts [hunk 5]
```diff
@@ -42,6 +42,8 @@ const ServerConfigDefaults: Partial<IServerConfig> = {
     isoformOverrideSource: 'uniprot',
     show_hotspot: true,
     show_oncokb: true,
+    show_cbioportal: true,
```
Comment (alisman) [inline src/config/serverConfigDefaults.ts:45]: this is obviously a long standing issue but we need to at very least improve specificity of propery names.  like, oncoprint_show_cbioportal_annotations?

File: src/shared/components/driverAnnotations/DriverAnnotationControls.tsx [hunk 6]
```diff
@@ -244,69 +245,74 @@ export default class DriverAnnotationControls extends React.Component<
...
-                            {this.props.handlers
-                                .onChangeAnnotateCBioPortalInputValue && (
-                                <div className="checkbox">
-                                    <label>
-                                        <input
-                                            type="checkbox"
-                                            value={EVENT_KEY.annotateCBioPortal}
-                                            checked={
+                            {getServerConfig().show_cbioportal &&
+                                this.props.handlers
```
Comment (alisman) [inline src/shared/components/driverAnnotations/DriverAnnotationControls.tsx:249]: odd that it needs to check for existnence of the onChange handler.  is it really possible that it wouldn't exist?

File: src/config/IAppConfig.ts [hunk 7]
```diff
@@ -66,6 +66,8 @@ export interface IServerConfig {
     api_cache_limit: number;
     show_hotspot: boolean | undefined;
     show_oncokb: boolean;
+    show_cbioportal: boolean;
```
Comment (adamabeshouse) [inline src/config/IAppConfig.ts:69]: i think this should be `oncoprint_show_cbioportal_annot` and `oncoprint_show_cosmic_annot`

File: src/shared/components/driverAnnotations/DriverAnnotationControls.tsx (hunk 6 above)
Comment (adamabeshouse) [reply src/shared/components/driverAnnotations/DriverAnnotationControls.tsx:249]: @alisman i think this is related to reusing this controls component in oncoprint and in oncoprinter

**Review by adamabeshouse — CHANGES_REQUESTED** [review]
mostly good, just some naming issues

**Pull request closed without merging**